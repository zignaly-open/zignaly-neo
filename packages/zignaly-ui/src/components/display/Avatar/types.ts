export enum AvatarSizes {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
  XLARGE = "x-large",
  XXLARGE = "xx-large",
}

export type AvatarTypeProps = {
  size?: AvatarSizes | AvatarSizes[keyof AvatarSizes];
  alt?: string;
  hash?: string | any;
  image?: string | any;
  id?: string;
};
