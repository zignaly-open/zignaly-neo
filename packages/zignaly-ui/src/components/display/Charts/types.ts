export interface ChartsProps {
  readonly variant: keyof typeof ChartVariations;
  readonly yAxis?: "left" | "right";
  readonly data: AxisFormat[];
  midLine?: boolean | null;
}

export interface AxisFormat {
  x: number | string;
  y: number;
}

export interface ChartInput {
  data: AxisFormat[];
  isGreen: boolean;
  midLine?: boolean | null;
}

export const ChartVariations = {
  small: "small",
  large: "large",
};

export const largeStyle = {
  axisLabel: {
    fontSize: 20,
    padding: 30,
    fill: "white",
    fontFamily: "Avenir Next",
    letterSpacing: 0.55,
    lineHeight: 16,
  },
  tickLabels: {
    fontSize: 11,
    padding: 5,
    fill: "white",
    fontFamily: "Avenir Next",
    letterSpacing: 0.55,
    lineHeight: 16,
  },
};
