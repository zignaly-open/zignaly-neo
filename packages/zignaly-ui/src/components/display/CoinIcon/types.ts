export enum CoinSizes {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
}

export type CoinTypeProps = {
  size?: CoinSizes | any;
  name: string;
  coin: string;
  className?: string | any;
};
