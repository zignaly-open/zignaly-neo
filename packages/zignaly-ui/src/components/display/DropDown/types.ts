import { PopoverOrigin, PopoverPosition } from "@mui/material/Popover/Popover";

type Option = {
  id?: string;
  label?: string | JSX.Element;
  onClick?: () => void;
  href?: string;
  target?: "_blank";
  active?: boolean;
};

export type DropDownOption = Option & {
  id?: string;
  children?: Option[];
  element?: JSX.Element;
  separator?: boolean;
};

export type DropDownProps = {
  options: DropDownOption[];
  component: ({ open }: { open: boolean }) => JSX.Element;
  anchorOrigin?: PopoverOrigin;
  anchorPosition?: PopoverPosition;
  transformOrigin?: PopoverOrigin;
};

export type DropDownHandle = {
  closeDropDown?: () => void;
};
