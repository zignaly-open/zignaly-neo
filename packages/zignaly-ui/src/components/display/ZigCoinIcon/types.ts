export const COIN_SIZES = {
  Small: "small",
  Medium: "medium",
  Large: "large",
} as const;

type CoinSizesKeys = keyof typeof COIN_SIZES;
export type CoinSizes = typeof COIN_SIZES[CoinSizesKeys];

export const sizes = {
  [COIN_SIZES.Small]: 24,
  [COIN_SIZES.Medium]: 36,
  [COIN_SIZES.Large]: 42,
};

export type ZIGCoinIconProps = {
  size?: CoinSizes | number;
  coin: string;
  className?: string;
  bucket?: string;
  id?: string;
};
