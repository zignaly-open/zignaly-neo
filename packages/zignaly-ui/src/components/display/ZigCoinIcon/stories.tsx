import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import ZigCoinIcon from "./index";

export default {
  title: "Display/ZigCoinIcon",
  component: ZigCoinIcon,
  argTypes: {
    size: {
      control: "select",
      default: "medium",
      options: ["small", "medium", "large"],
    },
    coin: {
      control: "text",
    },
  },
} as ComponentMeta<typeof ZigCoinIcon>;

const Template: ComponentStory<typeof ZigCoinIcon> = (args) => <ZigCoinIcon {...args} />;

export const EtherIconStory = Template.bind({});
EtherIconStory.args = {
  coin: "ETH",
};

export const BitcoinIconStory = Template.bind({});
BitcoinIconStory.args = {
  coin: "BTC",
};

export const TetherIconStory = Template.bind({});
TetherIconStory.args = {
  coin: "USDT",
};
