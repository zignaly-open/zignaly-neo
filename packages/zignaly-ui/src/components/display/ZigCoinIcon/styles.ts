import { styled } from "@mui/system";

export const Icon = styled("img")<{ size: number }>`
  ${({ size }) => `
    width: ${size}px;
    height: ${size}px;
  `}
  border-radius: 50%;
`;

export const Placeholder = styled(Icon)<{ size: number }>`
  display: flex;
  justify-content: center;
  align-items: center;
  background: #324054;
  font-size: 16px;
`;
