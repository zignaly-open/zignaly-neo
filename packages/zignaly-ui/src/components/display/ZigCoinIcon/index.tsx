import React, { useCallback, useEffect, useState } from "react";
import { Icon, Placeholder } from "./styles";
import { COIN_SIZES, sizes } from "./types";
import type { ZIGCoinIconProps, CoinSizes } from "./types";
import { ZignalyIcon } from "../../../icons";

const ZigCoinIcon = ({
  size = COIN_SIZES.Medium,
  coin,
  className = "",
  bucket = "coins-binance",
  id,
}: ZIGCoinIconProps) => {
  const s = typeof size === "number" ? size : sizes[size];
  const [src, setSrc] = useState("");
  const onError = useCallback(() => setSrc(""), []);
  useEffect(() => {
    setSrc(
      `https://res.cloudinary.com/zignaly/image/upload/c_scale,w_${s},h_${s},r_max/${bucket}/${coin}`,
    );
  }, [coin]);
  if (!coin) return null;

  return src ? (
    <Icon src={src} alt={coin} size={s} className={className} onError={onError} id={id} />
  ) : coin.toLowerCase() === "zig" ? (
    <ZignalyIcon width={sizes[size as CoinSizes]} height={sizes[size as CoinSizes]} id={id} />
  ) : (
    <Placeholder size={s} className={className} as="div" id={id}>
      {coin[0]}
    </Placeholder>
  );
};

export type { CoinSizes };
export default ZigCoinIcon;
