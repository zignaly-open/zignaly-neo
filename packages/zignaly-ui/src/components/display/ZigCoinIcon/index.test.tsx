import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { screen } from "@testing-library/react";
import ZigCoinIcon from ".";
import { COIN_SIZES, sizes } from "./types";

describe("components/display/ZigCoinIcon", () => {
  const mockCoin = "btc";

  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(<ZigCoinIcon coin={mockCoin} />);
    const coinIcon = screen.getByAltText(mockCoin);
    expect(coinIcon).toBeVisible();
    expect(container).toMatchSnapshot();
  });

  it("should set the correct source URL when coin prop changes", () => {
    const s = sizes[COIN_SIZES.Medium];
    const { rerender } = renderWithProvidersUi(<ZigCoinIcon coin={mockCoin} />);
    const coinIcon = screen.getByAltText(mockCoin);
    const expectedSrc = `https://res.cloudinary.com/zignaly/image/upload/c_scale,w_${s},h_${s},r_max/coins-binance/${mockCoin}`;
    expect(coinIcon).toHaveAttribute("src", expectedSrc);

    const newMockCoin = "eth";
    rerender(<ZigCoinIcon coin={newMockCoin} />);
    const newCoinIcon = screen.getByAltText(newMockCoin);
    const newExpectedSrc = `https://res.cloudinary.com/zignaly/image/upload/c_scale,w_${s},h_${s},r_max/coins-binance/${newMockCoin}`;
    expect(newCoinIcon).toHaveAttribute("src", newExpectedSrc);
  });

  it("should not render if coin prop is falsy", () => {
    renderWithProvidersUi(<ZigCoinIcon coin={""} />);
    const coinIcon = screen.queryByAltText("btc");
    expect(coinIcon).not.toBeInTheDocument();
  });

  it("should render ZignalyIcon when coin is 'zig'", () => {
    renderWithProvidersUi(<ZigCoinIcon coin="zig" />);
    const zignalyIcon = screen.queryByAltText("zig");
    expect(zignalyIcon).toBeInTheDocument();
  });
});
