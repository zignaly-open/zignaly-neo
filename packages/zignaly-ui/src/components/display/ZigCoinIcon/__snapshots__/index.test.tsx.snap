// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/display/ZigCoinIcon should render without crashing 1`] = `
<div>
  <img
    alt="btc"
    class=" css-1edqgwj"
    size="36"
    src="https://res.cloudinary.com/zignaly/image/upload/c_scale,w_36,h_36,r_max/coins-binance/btc"
  />
</div>
`;
